File: leetcode/Algorithms/242.ValidAnagram/Solution.java
public class Solution {
    public boolean isAnagram(String s, String t) {
        int[] cache = new int[26];
        for (int i = 0; i < s.length(); i++) cache[s.charAt(i) - 'a']++;
        for (int i = 0; i < t.length(); i++) cache[t.charAt(i) - 'a']--;
        for (int i : cache) if (i != 0) return false;
        return true;
    }
}

File: leetcode/Algorithms/242.ValidAnagram/Solution.c
bool isAnagram(char* s, char* t) {
    int cache[26] = {0};
    for (; *s!='\0'; s++) cache[*s - 'a']++;
    for (; *t!='\0'; t++) cache[*t - 'a']--;
    for (int i = 0; i < 26; i++) {
        if (cache[i] != 0) return false;
    }
    return true;
}
